vision/module/region_of_interest.py

```diff
@@ -0,0 +1,72 @@
+"""
+region_of_interest will
+given the depth value (float) in millimeters
+
+The __main__ of this file acts as a driver for testing region_of_interest
+"""
+import numpy as np
+import argparse
+
+# Constant
+MODULE_HEIGHT = 76.2  # mm
+MODULE_WIDTH = 50.8  # mm
+VERTICAL_FOV = 57  # degrees?
+HORIZONTAL_FOV = 86  # degrees
+VERTICAL_RES = 1080  # p
+HORIZONTAL_RES = 1920  # p
+
+# ratio of the measured region_of_interest, use from 0 to 1
+#   this should be used to ensure that the region_of_interest is trustworthy despite tilt
+PADDING_CONSTANT = .85
+
+
+def region_of_interest(depth_val, center):
+    """
+    Finds region of interest of the module in frame
+
+    Parameters
+    ----------
+    depth_val: float
+        Measured value for the depth of the module from the camera
+    center: integer tuple
+    """
+    x_pos, y_pos = center
+
+    # vertical angle of the module in the frame, from the center of the module to the top of the module
+    vertical_region_angle = np.degrees(np.arctan((MODULE_HEIGHT / 2) / depth_val))
+
+    # horizontal angle of the module in the frame, from the center of the module to its horizontal edge
+    horizontal_region_angle = np.degrees(np.arctan((MODULE_WIDTH / 2) / depth_val))
+
+    # both ratios should be only for one side, such that you can + or - some proportional value to the center to create region of interest
+    vertical_angle_ratio = vertical_region_angle / (VERTICAL_FOV / 2)
+    horizontal_angle_ratio = horizontal_region_angle / (HORIZONTAL_FOV / 2)
+
+    vertical_image_portion = int((vertical_angle_ratio * VERTICAL_RES / 2) * PADDING_CONSTANT)
+    horizontal_image_portion = int((horizontal_angle_ratio * HORIZONTAL_RES / 2) * PADDING_CONSTANT)
+
+    return depth_frame[y_pos - vertical_image_portion:y_pos + vertical_image_portion, x_pos - horizontal_image_portion:x_pos + horizontal_image_portion]
```

Comment:
It appears that depth_frame has not been defined in this function. color_image and depth_image should be made parameters of the function.